osivalidator/osi_general_validator.py

```diff
@@ -159,140 +150,50 @@ def main():
         LOGGER.info(None, f"Pass the {max_timestep} first timesteps")
     else:
         LOGGER.info(None, "Pass all timesteps")
-        max_timestep = DATA.timestep_count
-
-    # Dividing in several blast to not overload the memory
-    max_timestep_blast = 0
-
-    while max_timestep_blast < max_timestep:
-        # Clear log queue
-        LOGS = MANAGER.list()
-
-        # Increment the max-timestep to analyze
-        max_timestep_blast += args.blast
-        first_of_blast = max_timestep_blast - args.blast
-        last_of_blast = min(max_timestep_blast, max_timestep)
-
-        # Cache messages
-        DATA.cache_messages_in_index_range(first_of_blast, last_of_blast)
-        MESSAGE_CACHE.update(DATA.message_cache)
-
-        if args.parallel:
-            # Launch parallel computation
-            # Recreate the pool
+        max_timestep = None
+
+    index = 0
+    total_length = os.path.getsize(args.data)
+    current_pos = 0
+
+    with tqdm(total=total_length, unit="B", unit_scale=True, unit_divisor=1024) as pbar:
+        for message in trace:
```

Comment:
Yeah, that is probably a left over of an iteration of the code where blast/parallelization was still handled differently, hence enumerate would not have worked (and sadly more advanced batching primitives are only now appearing in Python). I had thought that there was also a bad interaction with tqdm, and while this might have been true for some prior iteration of how we used tqdm, it is no longer the case, so we can switch back to enumerate.